ValidationTest fails

Fails on Windows

```java
private Validation<String, String> validateName(String name) {
  return CharSeq.of(name).replaceAll(validNameChars, "").transform(seq ->
   seq.isEmpty() ? Validation.valid(name)
   : Validation.invalid("Name contains invalid characters: '" + seq.distinct().sorted() + "'"));
}
```

Compilation fails with error

```
Error:(549, 77) java: incompatible types: inferred type does not conform to upper bound(s)
    inferred: javaslang.control.Validation<? extends java.lang.Object,? extends java.lang.Object>
    upper bound(s): javaslang.control.Validation<java.lang.String,java.lang.String>,java.lang.Object
```

My Java:
```
java version "1.8.0_73"
Java(TM) SE Runtime Environment (build 1.8.0_73-b02)
Java HotSpot(TM) 64-Bit Server VM (build 25.73-b02, mixed mode)
```

<code>&lt;String, String&gt;</code> fixes problem 
```java
        private Validation<String, String> validateName(String name) {
            return CharSeq.of(name).replaceAll(validNameChars, "").transform(seq ->
                    seq.isEmpty() ? Validation.<String, String>valid(name)
                            : Validation.<String, String>invalid("Name contains invalid characters: '" + seq.distinct().sorted() + "'"));
        }
```

/cc @enelson

No, this is feature (bug?) of JDK installed on my Windows 

```java
// fine
private Validation<String, String> validateTestCompiled(Option<String> opt) {
    final Function<Option<String>, Validation<String, String>> f = o -> o.isEmpty() ?
            Validation.valid("valid")
            : Validation.invalid("invalid");
    return opt.transform(f);
}

// compilation error
// [ERROR] /C:/Users/rsennov/javaslang/javaslang/src/test/java/javaslang/control/ValidationTest.java:[565,33] 
// incompatible types: inferred type does not conform to upper bound(s)
// inferred: javaslang.control.Validation<? extends java.lang.Object,? extends java.lang.Object>
// upper bound(s): javaslang.control.Validation<java.lang.String,java.lang.String>,java.lang.Object
private Validation<String, String> validateTestNotCompiled(Option<String> opt) {
    return opt.transform(o -> o.isEmpty() ?
            Validation.valid("valid")
            : Validation.invalid("invalid"));
}
```

This code also fails
```java
// [ERROR] /C:/Users/rsennov/javaslang/javaslang/src/test/java/javaslang/control/ValidationTest.java:[19,29] 
// incompatible types: inferred type does not conform to upper bound(s)
// inferred: javaslang.collection.HashMap<? extends java.lang.Object,? extends java.lang.Object>
// upper bound(s): javaslang.collection.Map<java.lang.String,java.lang.String>,java.lang.Object
private Map<String, String> mapTest(Option<Void> opt) {
    return opt.transform(o -> o.isEmpty()? HashMap.empty() : HashMap.of("1", "2"));
}
```

--- a/javaslang/src/test/java/javaslang/control/ValidationTest.java
+++ b/javaslang/src/test/java/javaslang/control/ValidationTest.java
@@ -549,8 +549,8 @@ public class ValidationTest extends AbstractValueTest {
 
         private Validation<String, String> validateName(String name) {
             return CharSeq.of(name).replaceAll(validNameChars, "").transform(seq ->
-                    seq.isEmpty() ? Validation.valid(name)
-                            : Validation.invalid("Name contains invalid characters: '" + seq.distinct().sorted() + "'"));
+                    seq.isEmpty() ? Validation.<String, String>valid(name)
+                            : Validation.<String, String>invalid("Name contains invalid characters: '" + seq.distinct().sorted() + "'"));
         }
 
         private Validation<String, Integer> validateAge(int age) {

